### DISCUSSION TITLE
get a post initialData from multiple sources with unknown query key

### DISCUSSION THREAD

**Discussion by k4mr4n** [TanStack/query#2627] [Q&A] [answered]
Hey 👋 

To read the initialData from the list of posts I've seen in this [link](https://github.com/learnwithjason/all-about-react-query/blob/main/src/hooks/usePost.js) that we should iterate over the cached posts.
Now imagine we have multiple sources for the posts and all of them have a dynamic queryKey (filter, sort, limit, ...) so I'm using a keyFactory for them.
with this approach, I do not have access to the exact key of the feed query. the only approach which comes to my mind is that to setQueryData for the post in advance on feed success event.

I want to know if is there any other more efficient approaches that you can suggest to me?.

**Comment by TkDodo** [comment, marked as answer]
You can use `queryClient.getQueriesData`, which fuzzily matches keys and try to find and set the post that you want.
